include/bluetooth/services/bms_srv.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#ifndef BT_GATT_BMS_SRV_H_
+#define BT_GATT_BMS_SRV_H_
+
+/**@file
+ * @defgroup bt_gatt_bms_srv Bond Management Service API
+ * @{
+ * @brief API for the Bond Management Service (BMS).
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdlib.h>
+#include <zephyr/types.h>
+
+/** @brief BMS server callback structure. */
+struct bt_gatt_bms_srv_cb {
+	/** Operation authorization callback
+	 *
+	 *  If this callback is set, support for BMS Authorization Code
+	 *  functionality is enabled. This function is called whenever
+	 *  an authorized operation is requested by the client.
+	 *
+	 *  @param auth_code     Authorization Code
+	 *  @param auth_code_len Length of Authorization Code
+	 *
+	 * @retval true   If the authorization was successful.
+	 * @retval false  Otherwise.
+	 */
+	bool (*operation_authorize)(const uint8_t *auth_code,
+				    uint16_t auth_code_len);
```

Comment:
Do we need some more context? I would think the conn object would be useful here. Possibly the authorize op (requesting, all, rest)?

In order to have it easier if arguments needs to be added in the future it would be best if the arguments was put into
the struct, otherwise adding arguments to the callback is an API breaker.

Reply:
Yes, I think that additional context would be useful here